src/macros.rs

```diff
@@ -32,6 +32,41 @@ macro_rules! load_yaml {
     };
 }
 
+/// Allows you to pull the licence from your Cargo.toml at compile time. If the `license` field is
+/// empty, then the `licence-field` is read. If both fields are empty, then `None` is returned.
+///
+/// # Examples
+///
+/// ```no_run
+/// # #[macro_use]
+/// # extern crate clap;
+/// # use clap::App;
+/// # fn main() {
+/// let m = App::new("app")
+///             .version(crate_license!())
+///             .get_matches();
+/// # }
+/// ```
+#[cfg(not(feature = "no_cargo"))]
+#[macro_export]
+macro_rules! crate_license {
+    () => {
+        {
+            let license = env!("CARGO_PKG_LICENSE");
```

Comment:
Make this a `mut` and then the code can be better instead of nested conditionals